### ISSUE TITLE
PanicException: assertion failed: i < (self.bits.len() << 3)

### ISSUE THREAD

**Issue by ztsweet** [pola-rs/polars#1098]
#### Are you using Python or Rust?
python
#### Which feature gates did you use?
groupby agg etc.
#### What version of polars are you using?
0.8.17
```python
import polars as pl
from datetime import datetime
print(pl.__version__)
# '0.8.17'
#udf
`
`
def compute_age() -> pl.Expr:
    # Date64 is time in ms
    ms_to_year = 1e3 * 3600 * 24 * 365
    return (
        pl.lit(datetime(2021, 1, 1)) - pl.col("birthday").str.parse_date(pl.Date32)
    ) / (ms_to_year)
##
def avg_birthday(gender: str) -> pl.Expr:
    return (
        compute_age()
        .filter(pl.col("gender") == gender)
        .mean()
        .alias(f"avg {gender} birthday")
    )
def f(gender: str) -> pl.Expr:
    ms_to_year = 1e3 * 3600 * 24 * 365
    new_col = (pl.lit(datetime(2021, 1, 1))- \
    pl.col('birthday').filter(pl.col('gender') == gender).str.parse_date(pl.Date32))/(ms_to_year)
    
    return (new_col.mean().alias(f'avg {gender} birthday'))
##
`
`
print(pdata_1[avg_birthday('M')])
# shape: (1, 1)
# ╭──────────────────╮
# │ avg M birthday   │
# │ ---              │
# │ f64              │
# ╞══════════════════╡
# │ 59.0931460255335 │
# ╰──────────────────╯
print(pdata_1[f('M')])
# shape: (1, 1)
# ╭──────────────────╮
# │ avg M birthday   │
# │ ---              │
# │ f64              │
# ╞══════════════════╡
# │ 59.0931460255335 │
# ╰──────────────────╯
print(pdata_1.groupby('state').agg([avg_birthday('M')]).head())
# shape: (5, 2)
# ╭───────┬────────────────────╮
# │ state ┆ avg M birthday     │
# │ ---   ┆ ---                │
# │ str   ┆ f64                │
# ╞═══════╪════════════════════╡
# │ "IN"  ┆ 54.49497716894977  │
# ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ "CT"  ┆ 63.43799086757991  │
# ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ "UT"  ┆ 59.11415525114155  │
# ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ "NE"  ┆ 54.108675799086754 │
# ├╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ "WY"  ┆ 68.49497716894977  │
# ╰───────┴────────────────────╯
print(pdata_1.groupby('state').agg([f('M')]).head())
## error
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
<ipython-input-152-0602b187dc8a> in <module>
----> 1 print(pdata_1.groupby('state').agg([f('M')]).head())

~/miniconda3/envs/test/lib/python3.8/site-packages/polars/eager/frame.py in agg(self, column_to_agg)
   2466             elif isinstance(column_to_agg[0], pl.Expr):
   2467                 return (
-> 2468                     wrap_df(self._df)
   2469                     .lazy()
   2470                     .groupby(self.by)

~/miniconda3/envs/test/lib/python3.8/site-packages/polars/lazy/frame.py in collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, string_cache, no_optimization)
    266             string_cache,
    267         )
--> 268         return pl.eager.frame.wrap_df(ldf.collect())
    269 
    270     def fetch(

PanicException: assertion failed: i < (self.bits.len() << 3)
`
```

**Comment by ritchie46** [comment]
Hey thanks for your issue report, but have you also got your input data?

**Comment by ritchie46** [comment]
Note that `parse_date` is deprecated. `strptime` is the recommended function.

**Comment by ztsweet** [comment]
@ritchie46 inpute data[legislators-current.csv](https://theunitedstates.io/congress-legislators/legislators-current.csv)

**Comment by ztsweet** [comment]
i guess the problem is  here
```
pdata_1 = pdata.select(['state','gender','birthday'])
print(pdata_1.groupby('state').get_group('VI').filter(pl.col('gender') == 'M'))
# shape: (0, 3)
# ╭───────┬────────┬──────────╮
# │ state ┆ gender ┆ birthday │
# │ ---   ┆ ---    ┆ ---      │
# │ str   ┆ str    ┆ str      │
# ╞═══════╪════════╪══════════╡
# ╰───────┴────────┴──────────╯
```

**Comment by ritchie46** [comment]
Yes, that's also what I expect, but it should not panic. I will fix this.

Another tip

```python
    pl.col('birthday').filter(pl.col('gender') == gender).str.parse_date(pl.Date32))/(ms_to_year)
```

There is a `dt` namespace that will help you to do do the conversion to year:

```python
    pl.col('birthday').filter(pl.col('gender') == gender).str.parse_date(pl.Date32)).dt.year()
```

**Comment by ztsweet** [comment]
thanks and expect the new version will fix the problem :-)

**Comment by ritchie46** [comment]
Fixed in #1101

**Comment by ritchie46** [comment]
fixed and released
